activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -169,10 +169,21 @@ def travel_to(date_or_time, with_usec: false)
           now = date_or_time.to_time.change(usec: 0)
         end
 
-        stubbed_time = Time.now if simple_stubs.stubbing(Time, :now)
-        simple_stubs.stub_object(Time, :now) { at(now.to_f) }
-        simple_stubs.stub_object(Date, :today) { jd(now.to_date.jd) }
-        simple_stubs.stub_object(DateTime, :now) { jd(now.to_date.jd, now.hour, now.min, now.sec, Rational(now.utc_offset, 86400)) }
+        stubs = simple_stubs
+        stubbed_time = Time.now if stubs.stubbing(Time, :now)
+        stubs.stub_object(Time, :now) { at(now.to_f) }
+
+        stubs.stub_object(Time, :new) do |*args, **options|
+          if args.empty? && options.empty?
+            at(now.to_f)
```

Comment:
I don't have the time to dog more that that tonight, but wouldn't just returining `now` be more correct here?

```ruby
>> Time.new
=> 2023-02-13 23:16:32.853392 +0100
>> Time.now
=> 2023-02-13 23:16:35.353374 +0100
>> Time.at(Time.now.to_f)
=> 2023-02-13 23:16:37 1953881/2097152 +0100
```

Reply:
Oh I see you got it from the previous code, hum, that's bad... `to_f` lose precision.

Reply:
If I remove `to_f`, all test cases still pass. Are there any implications of removing it?

For the stubbed method, I can just use already stubbed `Time.now` and the `to_f` can be fixed separately in one place, if needed.

Reply:
> Are there any implications of removing it?

Well, means we return the same instance for everyone, so yeah, we might leak state. e.g. `Time#utc` will mutate the instance.

So rather than just return `now`, it's really `now.dup` that should be returned I think, but would be worth doing a quick git blame to see if this was discussed. If somehow there a reason `dup` can't be used, that are better ways to clone a Time.

Reply:
Removed `to_f`. It was added in https://github.com/rails/rails/pull/44088. There was no discussion regarding it.

We need to use `at(now)` and not `now.dup` for preserving the original class - https://github.com/rails/rails/blob/adbb85be9ca0a876ae7b865a188c3db192732ab6/activesupport/test/time_travel_test.rb#L268-L281